test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -201,7 +201,20 @@ void UcxContext::request_release(void *request)
 void UcxContext::connect_callback(ucp_conn_request_h conn_req, void *arg)
 {
     UcxContext *self = reinterpret_cast<UcxContext*>(arg);
-    UCX_LOG << "got new connection request " << conn_req;
+    ucp_conn_request_attr_t conn_req_attr;
+
+    conn_req_attr.field_mask = UCP_CONN_REQUEST_ATTR_FIELD_CLIENT_ADDR;
+    ucs_status_t status = ucp_conn_request_query(conn_req, &conn_req_attr);
+    if (status == UCS_OK) {
+        UCX_LOG << "got new connection request "<< conn_req << " from client "
+                << UcxContext::sockaddr_str((const struct sockaddr*)
+                                            &conn_req_attr.client_address,
+                                            sizeof(conn_req_attr.client_address));
+    } else {
+        UCX_LOG << "ucp_conn_request_query() failed: " << ucs_status_string(status);
+        UCX_LOG << "got new connection request " << conn_req;
```

Comment:
maybe print this as one line?
`"got new connection request " << conn_req << ", "ucp_conn_request_query() failed (" << ucs_status_string(status) << ")";`